test/system/src/test/java/io/pravega/test/system/DynamicRestApiTest.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.system;
+
+import com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.admin.StreamManager;
+import io.pravega.client.stream.ScalingPolicy;
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.client.stream.impl.DefaultCredentials;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+
+import io.pravega.test.system.framework.Environment;
+import io.pravega.test.system.framework.SystemTestRunner;
+import io.pravega.test.system.framework.Utils;
+import io.pravega.test.system.framework.services.Service;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.glassfish.jersey.client.authentication.HttpAuthenticationFeature;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+import mesosphere.marathon.client.MarathonException;
+
+import static org.junit.Assert.assertEquals;
+import static javax.ws.rs.core.Response.Status.OK;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class DynamicRestApiTest extends AbstractSystemTest {
+
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(5 * 60);
+
+    private final Client client;
+    private WebTarget webTarget;
+    private String restServerURI;
+    private String resourceURl;
+
+    public DynamicRestApiTest() {
+
+        org.glassfish.jersey.client.ClientConfig clientConfig = new org.glassfish.jersey.client.ClientConfig();
+        clientConfig.register(JacksonJsonProvider.class);
+        clientConfig.property("sun.net.http.allowRestrictedHeaders", "true");
+
+        if (Utils.AUTH_ENABLED) {
+            HttpAuthenticationFeature feature = HttpAuthenticationFeature.basic("admin", "1111_aaaa");
```

Comment:
The credentials should be coming from a properties file instead of being hard-coded. As far as I remember, there is a properties file used in system tests that contain properties such as these.

Reply:
The properties file suggest that there is token based authentication used with `io.pravega.client.ClientConfig`, but not this glassfish `ClientConfig`.